### PR TITLE
Wip http java doc fixes

### PR DESCRIPTION
**Pull request by rkuhn**

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala [hunk 1]
```diff
@@ -215,25 +233,26 @@ private[akka] class StreamSupervisor(settings: ActorFlowMaterializerSettings) ex
...
+      case Collect(pf, _)             ⇒ (ActorInterpreter.props(settings, List(fusing.Collect(settings.supervisionDecider)(pf))), ())
+      case Scan(z, f, _)              ⇒ (ActorInterpreter.props(settings, List(fusing.Scan(z, f, settings.supervisionDecider))), ())
       case Expand(s, f, _)            ⇒ (ActorInterpreter.props(settings, List(fusing.Expand(s, f))), ())
-      case Conflate(s, f, _)          ⇒ (ActorInterpreter.props(settings, List(fusing.Conflate(s, f, att.settings(settings).supervisionDecider))), ())
+      case Conflate(s, f, _)          ⇒ (ActorInterpreter.props(settings, List(fusing.Conflate(s, f, settings.supervisionDecider))), ())
       case Buffer(n, s, _)            ⇒ (ActorInterpreter.props(settings, List(fusing.Buffer(n, s))), ())
-      case MapConcat(f, _)            ⇒ (ActorInterpreter.props(settings, List(fusing.MapConcat(f, att.settings(settings).supervisionDecider))), ())
+      case MapConcat(f, _)            ⇒ (ActorInterpreter.props(settings, List(fusing.MapConcat(f, settings.supervisionDecider))), ())
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala]: yeah, that extra indirection was unnecessary

File: akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala [hunk 2]
```diff
@@ -215,25 +233,26 @@ private[akka] class StreamSupervisor(settings: ActorFlowMaterializerSettings) ex
...
 private[akka] object ActorProcessorFactory {
   import akka.stream.impl.Stages._
+  import ActorFlowMaterializerImpl._
 
   def props(materializer: ActorFlowMaterializerImpl, op: StageModule, parentAttributes: OperationAttributes): (Props, Any) = {
     val att = parentAttributes and op.attributes
     // USE THIS TO AVOID CLOSING OVER THE MATERIALIZER BELOW
     // Also, otherwise the attributes will not affect the settings properly!
-    val settings = att.settings(materializer.settings)
+    val settings = calcSettings(att)(materializer.settings)
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala]: what does this buy?

File: akka-stream/src/main/scala/akka/stream/scaladsl/OperationAttributes.scala [hunk 3]
```diff
@@ -33,14 +32,6 @@ final case class OperationAttributes private (private val attributes: List[Opera
     case _          ⇒ "unknown-operation"
   }
 
-  private[akka] def settings: ActorFlowMaterializerSettings ⇒ ActorFlowMaterializerSettings =
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/scaladsl/OperationAttributes.scala]: Ah I see, you migrated this to the ActorFlowMat! good!

File: akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala (hunk 2 above)
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/impl/ActorFlowMaterializerImpl.scala]: forget it, I see now the removal from OpAttributes

File: akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala [hunk 4]
```diff
@@ -296,6 +298,46 @@ object FlowGraph {
...
+    def from[I, O](j: UniformFanOutShape[I, O]): ForwardOps[O] = new ForwardOps(findOut(delegate, j, 0))
+
+    def to[T](in: Inlet[T]): ReverseOps[T] = new ReverseOps(in)
+    def to[T, M](dst: Sink[T, M]): ReverseOps[T] = new ReverseOps(delegate.add(dst.asScala))
+    def to[T](dst: SinkShape[T]): ReverseOps[T] = new ReverseOps(dst.inlet)
+    def to[I, O](f: FlowShape[I, O]): ReverseOps[I] = new ReverseOps(f.inlet)
+    def to[I, O](j: UniformFanInShape[I, O]): ReverseOps[I] = new ReverseOps(findIn(delegate, j, 0))
+    def to[I, O](j: UniformFanOutShape[I, O]): ReverseOps[I] = new ReverseOps(j.in)
+
+    class ForwardOps[T](out: Outlet[T]) {
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala]: great!

File: akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala [hunk 5]
```diff
@@ -296,6 +298,46 @@ object FlowGraph {
...
+      def to[U](f: FlowShape[T, U]): Builder = { out ~> f; self }
+      def to[U](j: UniformFanInShape[T, U]): Builder = { out ~> j; self }
+      def to[U](j: UniformFanOutShape[T, U]): Builder = { out ~> j; self }
+      def via[U, M](f: Flow[T, U, M]): ForwardOps[U] = from((out ~> f.asScala).outlet)
+      def via[U](f: FlowShape[T, U]): ForwardOps[U] = from((out ~> f).outlet)
+      def via[U](j: UniformFanInShape[T, U]): ForwardOps[U] = from((out ~> j).outlet)
+      def via[U](j: UniformFanOutShape[T, U]): ForwardOps[U] = from((out ~> j).outlet)
+    }
+
+    class ReverseOps[T](out: Inlet[T]) {
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala]: great!

File: akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/FlowGraphDocTest.java [hunk 6]
```diff
@@ -50,26 +43,33 @@ public static void tearDown() {
...
+    final RunnableFlow<Future<List<String>>> result = FlowGraph.factory()
+        .closed(
+            sink,
+            (b, out) -> {
+              final UniformFanOutShape<Integer, Integer> bcast =
+                  b.graph(Broadcast.create(2));
+              final UniformFanInShape<Integer, Integer> merge =
+                  b.graph(Merge.create(2));
+
+              b.from(in).via(f1).via(bcast).via(f2).via(merge)
```
Comment (drewhk) [inline akka-samples/akka-docs-java-lambda/src/test/java/docs/stream/FlowGraphDocTest.java]: yaaay!

File: akka-stream/src/main/scala/akka/stream/FlowMaterializer.scala [hunk 7]
```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream
+
+import scala.concurrent.ExecutionContextExecutor
+import akka.japi
+
+abstract class FlowMaterializer {
```
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/FlowMaterializer.scala]: yep, good that you migrated this

File: akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala (hunk 5 above)
Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala]: hmm. Actually this ReverseOps feels wrong, it does not read as nicely as the arrows in the scaladsl. Aren't they confusing?

Comment (drewhk) [inline akka-stream/src/main/scala/akka/stream/javadsl/Graph.scala]: I mean, we have no such reverse ops in the simple Flow API, you always go left-to-right from Sources, but this allows

**Comment by drewhk** [comment]
LGTM, except the ReverseOps which I think go against the philosophy of the normal Flow API.

**Comment by rkuhn** [comment]
We should try out the reverse direction, it was just a straight port to get the DSLs “on par”, but if it does not work out then I’ll remove that part of course. The best place to try this is the cycles spec.

**Comment by drewhk** [comment]
My point is that if we include such API then why don't we include it for Sink?

**Comment by rkuhn** [comment]
WDYM? `b.to(sink)...` works already, or am I missing something?

**Comment by drewhk** [comment]
I meant the ordinary Flow API which is strictly source-to-sink, but cannot go in the other direction.

**Comment by drewhk** [comment]
LGTM

**Comment by drewhk** [comment]
Ah, small fixes incoming :)

**Pull request merged**